fence/agents/powerman/fence_powerman.py

```diff
@@ -0,0 +1,300 @@
+			logging.error("PowerMan Error: Popen: invalid arguments: %s\n", e)
+			return -1
+		if ret_code < 0:
+			# there was an error with the command
+			return ret_code
+		else:
+			state = {}
+			for line in result[0].split('\n'):
+				if len(line) > 2:
+					fields = line.split(':')
+					if len(fields) == 2:
+						state[fields[0]] = (fields[0],fields[1])
+			return state
+
+	def query(self, host):
+		cmd = ["--query", host]
+		try:
+			result, ret_code = self._run(cmd, True)
+		except OSError as e:
+			logging.error("PowerMan Error: The command '%s' failed: %s\n", cmd, e)
+			return -1
+		except ValueError as e:
+			logging.error("PowerMan Error: Popen: invalid arguments: %s\n", e)
+			return -1
+		if ret_code < 0:
+			# there was an error with the command
+			return ret_code
+		else:
+			res = result.split('\n')
+			res = [r.split() for r in res]
+			# find the host in command's returned output
+			for lst in res:
+				if lst[0] == 'No' and lst[1] == 'such' and lst[2] == 'nodes:':
+					return -1
+				if host in lst:
+					return lst[0][:-1] # lst[0] would be 'off:'-- this removes the colon
+			# host isn't in the output
+			return -1
+
+
+def get_power_status(conn, options):
+	logging.debug("get_power_status function:\noptions: %s\n", str(options))
+	pm = PowerMan(options['--ip'], options['--ipport'])
+	# if Pacemaker is checking the status of the Powerman server...
+	if options['--action'] == 'monitor':
+		if pm.is_running():
+			logging.debug("Powerman is running\n")
+			return "on"
+		logging.debug("Powerman is NOT running\n")
+		return "off"
+	else:
+		status = pm.query(options['--plug'])
+		if isinstance(int, type(status)):
+			# query only returns ints on error
+			logging.error("get_power_status: query returned %s\n", str(status))
+			fail(EC_STATUS)
+		return status
+
+
+def set_power_status(conn, options):
+	logging.debug("set_power_status function:\noptions: %s", str(options))
+	pm = PowerMan(options['--ip'], options['--ipport'])
+
+	action = options["--action"]
+	if action == "on":
+		pm.on(options['--plug'])
+	elif action == "off":
+		pm.off(options['--plug'])
+
+	return
+
+
+def reboot(conn, options):
+	logging.debug("reboot function:\noptions: %s", str(options))
+	pm = PowerMan(options['--ip'], options['--ipport'])
+	res = pm.off(options['--plug'])
+	if res < 0:
+		logging.error("reboot: power off failed.\n")
+		return False
+	time.sleep(2)
+	res = pm.on(options['--plug'])
+	if res < 0:
+		logging.error("reboot: power on failed.\n")
+		return False
+	return True
+
+
+def get_list(conn, options):
+	logging.debug("get_list function:\noptions: %s", str(options))
+	pm = PowerMan(options['--ip'], options['--ipport'])
+
+	outlets = pm.list()
+	logging.debug("get_list outlets.keys: %s", str(outlets.keys()))
+	return outlets
+
+
+def define_new_opts():
+	"""add elements to all_opt dict if you need to define new options"""
+	pass
+
+
+def main():
+	device_opt = [
+		'ipaddr',
+		'no_password',
+		'no_login',
+	]
+
+	raw_args=sys.argv
```

Comment:
This is something, what you don't want to have in the code. While I agree that it should be fine to see raw input and it can be added into fencing library, your solution did not count with using STDIN as an input method. And stdin is the standard way how to handle arguments in cluster environment.